### PR TITLE
Add maven info

### PR DESCRIPTION
**Pull request by lineargraph**
## What
Describe what this pull request does, including technical details, screenshots, links to discord, etc.

## Changelog Technical Details
+ Add maven publication details, allowing other mods to integrate more easily with Skyhanni. - !nea

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -257,3 +258,31 @@ val compileTestKotlin: KotlinCompile by tasks
...
+publishing.publications {
+    create<MavenPublication>("maven") {
+        artifact(tasks.remapJar)
+        artifact(sourcesJar) { classifier = "sources" }
+        pom {
+            name.set("SkyHanni")
+            licenses {
+                license {
+                    name.set("GNU Lesser General Public License")
+                    url.set("https://github.com/hannibal002/SkyHanni/blob/master/COPYING.LESSER")
```
Comment (hannibal002) [inline build.gradle.kts]: This url is 404 for me. wrong branch name?

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Just to be sure: this will upload the jar to maven? Or what exactly is this doing?

File: build.gradle.kts (hunk 1 above)
Comment (lineargraph) [reply build.gradle.kts]: OOps

**Comment by lineargraph** [comment]
> Just to be sure: this will upload the jar to maven? Or what exactly is this doing?

This uploads the JAR nowhere. But it encodes enough maven info in the project for https://jitpack.io/ to build the JAR and serve it from there. If you want to, i could probably publish the JAR to the neu maven, but i didn't feel the need for that.

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**